### PR TITLE
Document release conventions

### PR DESCRIPTION
**Pull request by gvelez17**
# Document CAS release conventions

## Description

As the team grows we should have explicit norms around building and deploying CAS to Clay and Production

It is also desireable to reduce the cycle time for releases if possible to do so safely.

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

Documentation only, n/a

## Definition of Done

Before submitting this PR, please make sure:

- [ ] The work addresses the description and outcomes in the issue
- [ ] I have added relevant tests for new or updated functionality
- [ ] My code follows conventions, is well commented, and easy to understand
- [ ] My code builds and tests pass without any errors or warnings
- [x ] I have tagged the relevant reviewers
- [ ] I have updated the READMEs of affected packages
- [x ] I have made corresponding changes to the documentation
- [ ] The changes have been communicated to interested parties

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

**Comment by gvelez17** [comment]
pinged everyone in case you have opinions, only need one actual review

**Review by smrz2001 — APPROVED** [review]
🥳

**Review by ukstv — APPROVED** [review]
Nice!

File: docs/DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Github Actions
+
+The available github actions are controlled by the files in [.github/workflows](../.github/workflows)
+
+The *Publish Release* action causes github to merge changes to the selected target and to run `npm release`,
+which in turn triggers a separate workflow that deploys to the 3box infrastructure.
+
+Available targets are listed as "Type of publish"
+
+ - rc     : will merge to release-candidate and deploy to Clay
```
Comment (stbrody) [inline docs/DEPLOYMENT.md:16]:
```suggestion
 - rc     : will merge develop to release-candidate and deploy to Clay
```

File: docs/DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ - main   : will merge release-candidate to main and deploy to Mainnet
+ - both   : will merge to release-candidate and mainnet and deploy to both Clay and Mainnet
```
Comment (stbrody) [inline docs/DEPLOYMENT.md:18]:
```suggestion
 - both   : will first merge release-candidate to main and deploy to Mainnet, then merge develop to release-candidate and deploy to Clay
```

File: docs/DEPLOYMENT.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ - hotfix : will build the release from the current main branch
```
Comment (stbrody) [inline docs/DEPLOYMENT.md:19]:
```suggestion
 - hotfix : will build the release from the current main branch without doing any cross-branch merging.
```

**Pull request merged**